<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Marie Saitou" />


<title>Cross-Ancestry LCLs eQTL (1) Data Preparation</title>

<script src="site_libs/header-attrs-2.26/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">myproject</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/mariesaitou/myproject">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Cross-Ancestry LCLs eQTL (1) Data
Preparation</h1>
<h4 class="author">Marie Saitou</h4>
<h4 class="date">7/1/2021</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-09-21
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>cross_ancestory_LCL/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200813code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20200813)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200813code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200813)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcommariesaitoumyprojecttree20c4949c5ccbf1e0c013d8d10bda5856b36a9ef4targetblank20c4949a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/mariesaitou/myproject/tree/20c4949c5ccbf1e0c013d8d10bda5856b36a9ef4" target="_blank">20c4949</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcommariesaitoumyprojecttree20c4949c5ccbf1e0c013d8d10bda5856b36a9ef4targetblank20c4949a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/mariesaitou/myproject/tree/20c4949c5ccbf1e0c013d8d10bda5856b36a9ef4" target="_blank">20c4949</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.RData
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/11.24.2020.2.png
    Ignored:    analysis/11.24.2020.3.png
    Ignored:    analysis/11.24.2020.4.png
    Ignored:    analysis/11.24.2020.5.png
    Ignored:    analysis/11.24.2020.6.png
    Ignored:    analysis/11.24.2020.7.png
    Ignored:    analysis/11.24.2020.8.png
    Ignored:    analysis/11.24.2020.9.png
    Ignored:    genotype/

Untracked files:
    Untracked:  analysis/1.21.WIP.Rmd
    Untracked:  analysis/12.WIP.Rmd
    Untracked:  analysis/first-analysis.Rmd alias
    Untracked:  getfastqtest2.csv

Unstaged changes:
    Modified:   analysis/correl.Rmd
    Modified:   analysis/manuscript2.Rmd
    Modified:   analysis/manuscript3.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/manuscript1.Rmd</code>) and HTML
(<code>docs/manuscript1.html</code>) files. If you’ve configured a
remote Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mariesaitou/myproject/blob/20c4949c5ccbf1e0c013d8d10bda5856b36a9ef4/analysis/manuscript1.Rmd" target="_blank">20c4949</a>
</td>
<td>
mariesaitou
</td>
<td>
2024-09-21
</td>
<td>
workflowr::wflow_publish("analysis/manuscript1.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mariesaitou/myproject/9474f66537f8fca96f3521df850430bf38e23111/docs/manuscript1.html" target="_blank">9474f66</a>
</td>
<td>
mariesaitou
</td>
<td>
2022-04-11
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mariesaitou/myproject/blob/c6a488f0b70e081323923379084ed7da6ace3807/analysis/manuscript1.Rmd" target="_blank">c6a488f</a>
</td>
<td>
mariesaitou
</td>
<td>
2022-04-11
</td>
<td>
Update the files for myproject
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mariesaitou/myproject/d23b1beb8e50a7d5566a2c1dd1f4064f9456a9c0/docs/manuscript1.html" target="_blank">d23b1be</a>
</td>
<td>
mariesaitou
</td>
<td>
2021-07-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mariesaitou/myproject/64421f6195b9c9ee98f59a1cbeddb2f0b1cfdb55/docs/manuscript1.html" target="_blank">64421f6</a>
</td>
<td>
mariesaitou
</td>
<td>
2021-07-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mariesaitou/myproject/d2e1bc0f1441b1f6f69108242e3d96968cd1de7c/docs/manuscript1.html" target="_blank">d2e1bc0</a>
</td>
<td>
mariesaitou
</td>
<td>
2021-07-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mariesaitou/myproject/2abb7f5482b35b04662bc938e761cf9592b2eac5/docs/manuscript1.html" target="_blank">2abb7f5</a>
</td>
<td>
mariesaitou
</td>
<td>
2021-07-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mariesaitou/myproject/blob/90217b968b7ccdbb9447259db6d4ce01f5cc8750/analysis/manuscript1.Rmd" target="_blank">90217b9</a>
</td>
<td>
mariesaitou
</td>
<td>
2021-07-02
</td>
<td>
Publish the files for myproject
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="genotype-dataset" class="section level1">
<h1>Genotype Dataset</h1>
<div
id="download-the-genotype-dataset-from-the-1000-genomes-project-phase-3"
class="section level2">
<h2>Download the genotype dataset from the 1000 Genomes project phase
3</h2>
<pre class="bash"><code>
mkdir genotype
wget ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/release/20130502/ALL.chr{1..22}.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz
</code></pre>
</div>
<div id="clean-the-genotype-files-with-vcftools" class="section level2">
<h2>Clean the genotype files with vcftools</h2>
<pre class="bash"><code># keep the 455 individuals who are reported in both the 1000 Genome Project and Geuvadis
module load vcftools
for i in `seq 1 22`
do
   vcftools --gzvcf /genotype/phase3/ALL.chr$i.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep 1000g.ind.sample.txt --recode --out /genotype/phase3/chr$i.1000gphase3.455 
done

# exclude rare (less than 0.01 frequency) variants 
vcftools --vcf /genotype/phase3/chr$i.1000gphase3.455.recode.vcf --maf 0.01 --max-maf 0.99 --recode --out /genotype/phase3/chr$i.1000gphase3.455.0.01

# keep only bi-allelic variants
vcftools --vcf /genotype/phase3/chr$i.1000gphase3.455.0.01.recode.vcf --min-alleles 2 --max-alleles 2 --recode --out /genotype/phase3/chr$i.1000gphase3.455.0.01.biallelic
Identify covariates (genetic and non-genetic)

# Convert vcf file to plink format
plink --vcf chr22.1000gphase3.455.0.01.recode.vcf --make-bed --out chr22.1000gphase3.455.0.01.recode.bed

# Run pca
plink --bfile chr22.1000gphase3.455.0.01.recode.bed --pca --out</code></pre>
<pre class="bash"><code># Plot pca
library(ggplot2)
ggplot(data=pca, aes(V3,V4)) + geom_point()</code></pre>
</div>
</div>
<div id="gene-expression-dataset" class="section level1">
<h1>Gene Expression Dataset</h1>
<div id="gene-expression-quantification-with-kallisto"
class="section level2">
<h2>Gene expression quantification with Kallisto</h2>
<pre class="r"><code># Download the fastq dataset from the Geuvadis project
# sample information https://www.ebi.ac.uk/arrayexpress/files/E-GEUV-1/
# fastq files ftp://ftp.sra.ebi.ac.uk/vol1/fastq/
 
csvfile=getfastq.csv
for line in `cat ${csvfile} | grep -v ^#`
do
  url=`echo ${line} | cut -d &#39;,&#39; -f 2`
  wget ${url}
done

#FastQC - quality check

module load fastqc
fastqc *.fastq.gz -o /fastqc_out/

module load multiqc
multiqc . -f

#Kallisto - make index files - pseudomapping
module load kallisto
wget ftp://ftp.ensembl.org/pub/grch37/current/fasta/homo_sapiens/cdna/Homo_sapiens.GRCh37.cdna.all.fa.gz
gunzip Homo_sapiens.GRCh37.cdna.all.fa.gz
kallisto index -i human.GRCh37.cdna.all.idx Homo_sapiens.GRCh37.cdna.all.fa

wget https://github.com/pachterlab/kallisto-transcriptome-indices/releases/download/ensembl-96/homo_sapiens.tar.gz
gunzip Homo_sapiens
# homo_sapiens/transcriptome.idx
# Gene expression quantification for all samples with Kallisto

module load kallisto
csvfile=1000g.sample.csv
for line in `cat ${csvfile} | grep -v ^#`
do
  file=`echo ${line} | cut -d &#39;,&#39; -f 1`
  kallisto quant -i human.GRCh37.cdna.all.idx -o kallisto/${file}.kallisto -n 100 -t 32 fastq2/${file}.1.fastq.gz fastq2/${file}.2.fastq.gz
done</code></pre>
</div>
<div
id="generate-sample---gene-expression-matrix-file-from-the-kallisto-output"
class="section level2">
<h2>Generate “sample - gene expression” matrix file from the kallisto
output</h2>
<pre class="r"><code>library(tximport)
library(biomaRt)
update.packages()
install.packages(&quot;rlang&quot;)

## convert transcripts into genes
mart &lt;- useDataset(&quot;hsapiens_gene_ensembl&quot;, useMart(&quot;ensembl&quot;, host=&quot;grch37.ensembl.org&quot;))
df &lt;- read.table(&quot;kallisto/HG00096.kallisto/abundance.tsv&quot;, header = TRUE,sep = &quot;\t&quot;)
genes &lt;- df$target_id

G_list &lt;- getBM(filters= &quot;ensembl_transcript_id_version&quot;, attributes=c(&#39;ensembl_transcript_id_version&#39;,&#39;ensembl_gene_id&#39;),mart= mart,values=genes)
head(G_list)
write.csv(G_list, file = &quot;transcripts.to.genes2.csv&quot;, append = FALSE, quote = FALSE, sep = &quot;,&quot;,
          eol = &quot;\n&quot;, na = &quot;NA&quot;, dec = &quot;.&quot;, row.names = FALSE,
          col.names = FALSE, qmethod = &quot;double&quot;,
          fileEncoding = &quot;&quot;)


# summarize (gene expression) data from each individual
samples &lt;-read.csv(&quot;1000g.EUR.sample.csv&quot;,header=TRUE)
files &lt;- file.path(dir, &quot;kallisto&quot;, samples$kallisto, &quot;abundance.tsv&quot;)
names(files) &lt;- paste0(samples$X)
all(file.exists(files))
txi &lt;- tximport(files, type = &quot;kallisto&quot;, tx2gene = tx2gene, txOut = FALSE, ignoreAfterBar = TRUE)

write.table(txi$est_counts,sep=&quot;\t&quot;, file = &quot;EUR.counts.tsv&quot;)
write.table(txi$tpm,sep=&quot;\t&quot;, file = &quot;EUR.TPM.tsv&quot;)</code></pre>
</div>
<div id="add-gene-annotations-to-the-gene-expression-matrix-file"
class="section level2">
<h2>Add gene annotations to the gene expression matrix file</h2>
<pre class="r"><code>library(&#39;biomaRt&#39;)
mart &lt;- useDataset(&quot;hsapiens_gene_ensembl&quot;, useMart(&quot;ensembl&quot;, host=&quot;grch37.ensembl.org&quot;))
df &lt;- read.csv(file.choose(), header = T, sep = &quot;,&quot;) 
gene &lt;- df$gene
G_list &lt;- getBM(filters= &quot;ensembl_gene_id&quot;, attributes=c(&#39;ensembl_gene_id&#39;, &#39;hgnc_symbol&#39;,&#39;description&#39;,&#39;chromosome_name&#39;,&#39;start_position&#39;,&#39;end_position&#39;),mart= mart,values=gene)
G_list2&lt;-merge(df,G_list,by.x=&quot;gene&quot;,by.y=&quot;ensembl_gene_id&quot;,all.x=T)
write.csv(G_list2, file = &quot;EUR.gene.TPM.csv&quot;)


## only genes with CPM (counts per million)&gt;0.5 in more than half of the total samples 
## are kept for further analysis to avoid false positives in the eQTL
d1  &lt;- read.table(&quot;EUR.counts.tsv&quot;, header = T,row.names=1,sep = &quot;\t&quot;)
d1_scaled &lt;- apply(d1, 2, function(x){x/sum(x)*1000000})
d1_cleaned&lt;-d1_scaled[rowSums(d1_scaled &gt; 0.5) &gt; length(d1[1,])/2, ]

write.csv(d1_cleaned, file = &quot;EUR.CPM.survived.csv&quot;)</code></pre>
</div>
<div id="scale-the-gene-expression-matrix" class="section level2">
<h2>Scale the gene expression matrix</h2>
<pre class="r"><code>matall&lt;-read.table(file = &#39;Yoruba.TPM.filtered.tsv&#39;, sep = &#39;\t&#39;, header = T,row.names=1)
matall&lt;-as.matrix(matall,row.names=1,header=T)
head(matall)
matfilter &lt;- matall[rowSums(matall) &gt; 90,]
hist(mat[,1])
hist(mat[1,],breaks=20)
mat&lt;-normalize.quantiles(matfilter)
mat&lt;-scale (mat)
tmat&lt;-t(mat)
stmat&lt;-scale (tmat)
mat1&lt;-t(stmat)
head(mat1)
hist(mat1[,1])
hist(mat1[1,],breaks=40)
class(mat)
colnames(mat)&lt;-matal
write.csv(mat1, file = &quot;Yoruba.TPM.scaled.csv&quot;)</code></pre>
</div>
<div id="identify-genetic-covariates" class="section level2">
<h2>Identify genetic covariates</h2>
<pre class="bash"><code># Convert vcf file to plink format
plink --vcf chr22.1000gphase3.455.0.01.recode.vcf --make-bed --out chr22.1000gphase3.455.0.01.recode.bed

# Run pca
plink --bfile chr22.1000gphase3.455.0.01.recode.bed --pca --out

# Plot pca
library(ggplot2)
ggplot(data=pca, aes(V3,V4)) + geom_point()
</code></pre>
</div>
<div id="identify-sarrogate-variables" class="section level2">
<h2>Identify sarrogate variables</h2>
<pre class="r"><code>#  surrogate variable analysis
library(sva)
mm &lt;- model.matrix(~ population, colData(ddsTxi))
mm0 &lt;- model.matrix(~ 1, colData(ddsTxi))
norm.cts &lt;- norm.cts[rowSums(norm.cts) &gt; 10,]
fit &lt;- svaseq(norm.cts, mod=mm, mod0=mm0)
# 50 SVs are found. I use 7 SVs
sva7 = sva(norm.cts , mm, mm0, n.sv=7)
write.csv(sva7$sv[,1:7], file = &quot;sva7.csv&quot;)</code></pre>
</div>
</div>
<div id="differentially-expressed-gene-analysis" class="section level1">
<h1>Differentially Expressed Gene Analysis</h1>
<pre class="r"><code>library(&quot;DESeq2&quot;)
colData &lt;-read.csv(&quot;1000g.sample.csv&quot;,header = T,row.names=1)
class(colData$date)
colData$date1&lt;-factor(colData$date)


ddsTxi &lt;- DESeqDataSetFromTximport(txi,
                                   colData = colData,
                                   design = ~ sex+population+performer)

colData &lt;-read.csv(&quot;1000g.EUR.YRI.csv&quot;,header = T,row.names=1)

ddsEurYri &lt;- DESeqDataSetFromTximport(txi,
                                   colData = colData,
                                   design = ~ sex+population+V1+V2+V3+V4+V5+V6+V7)

keep &lt;- rowsum(counds(ddsEurYri)) &gt;=455
ddsEurYri &lt;- ddsEurYri[keep,]
ddsEY&lt;-DESeq(ddsEurYri)
deg &lt;- results(ddsEY, contrast=c(&quot;population&quot;,&quot;EUR&quot;,&quot;Yoruba&quot;))
write.csv(deg, file = &quot;E-GEUV_EUR_Yoruba.csv&quot;)</code></pre>
</div>
<div id="cis-eqtl-mapping-analysis" class="section level1">
<h1>Cis-eQTL Mapping Analysis</h1>
<pre class="r"><code># prepare input bed.gz and index files
module load bedtools
bedtools sort -i British.gene.TPM.bed -header &gt; British.gene.TPM.sorted.bed
bgzip British.gene.TPM.sorted.bed &amp;&amp; tabix -p bed British.gene.TPM.sorted.bed.gz


# prepare input vcf.gz and index files
module load htslib 
module load tabix
for i in `seq 1 22`
do
    bgzip genotype/phase3/chr$i.1000gphase3.Yoruba.0.01.biallelic.recode.vcf  &amp;&amp; tabix -p vcf genotype/phase3/chr$i.1000gphase3.Yoruba.0.01.biallelic.recode.vcf.gz
done

for i in `seq 1 22`
do
    bgzip genotype/phase3/chr$i.1000gphase3.EUR.0.01.biallelic.recode.vcf  &amp;&amp; tabix -p vcf genotype/phase3/chr$i.1000gphase3.EUR.0.01.biallelic.recode.vcf.gz
done

# run fastQTL
## https://github.com/francois-a/fastqtl


for i in `seq 1 22`
do
  ./bin/fastQTL.static fastQTL --vcf /project2/xuanyao/marie/E-GEUV-1/genotype/phase3/chr$i.1000gphase3.EUR.0.01.biallelic.recode.vcf.gz --bed GEUV/EUR.gene.TPM.sorted.bed.gz --region $i:1-249250621  --threshold 0.001 --permute 1000 --out GEUV/EUR.chr$i.permute.0.001.txt --cov GEUV/EUR.cov.txt --normal
done


for i in `seq 1 22`
do
  ./bin/fastQTL.static fastQTL --vcf /project2/xuanyao/marie/E-GEUV-1/genotype/phase3/chr$i.1000gphase3.Yoruba.0.01.biallelic.recode.vcf.gz --bed GEUV/Yoruba.gene.TPM.sorted.bed.gz --region $i:1-249250621  --threshold 0.001 --permute 1000 --out GEUV/Yoruba.chr$i.permute.0.001.txt --cov GEUV/Yoruba.cov.txt --normal
done</code></pre>
<div id="cleaning-and-formatting-the-eqtl-result"
class="section level2">
<h2>cleaning and formatting the eQTL result</h2>
<pre class="r"><code>setwd(&quot;/project2/xuanyao/marie/E-GEUV-1&quot;)
library(ggplot2)
library(&quot;ggrepel&quot;)
library(&quot;plyr&quot;)
library(gplots)
library(reshape2)


## remove multi-allelic SNPs reported as bi-allelic SNPs which confuse the result
EUR1&lt;-subset(EUR2, 
               EUR2$gene.SNP!=&quot;ENSG00000188659.rs542232278&quot;&amp;
               EUR2$gene.SNP!=&quot;ENSG00000255769.rs145926341&quot;&amp;
               EUR2$gene.SNP!=&quot;ENSG00000255769.rs371891811&quot;&amp;
               EUR2$gene.SNP!=&quot;ENSG00000259328.rs145926341&quot;&amp;
               EUR2$gene.SNP!=&quot;ENSG00000259323.rs145926341&quot;&amp;
               EUR2$gene.SNP!=&quot;ENSG00000259472.rs145926341&quot;&amp;
               EUR2$gene.SNP!=&quot;ENSG00000103942.rs1610794&quot;&amp;
               EUR2$gene.SNP!=&quot;ENSG00000103342.rs140839133&quot;&amp;
               EUR2$gene.SNP!=&quot;ENSG00000159202.rs77094622&quot;&amp;
               EUR2$gene.SNP!=&quot;ENSG00000189050.rs112549034&quot;&amp;
               EUR2$gene.SNP!=&quot;ENSG00000108592.rs138776605&quot;&amp;
               EUR2$gene.SNP!=&quot;ENSG00000256771.rs10650867&quot;&amp;
               EUR2$gene.SNP!=&quot;ENSG00000134330.rs139337028&quot;&amp;
               EUR2$gene.SNP!=&quot;ENSG00000125991.rs139252705&quot;&amp;
               EUR2$gene.SNP!=&quot;ENSG00000214078.rs142898689&quot;&amp;               
               EUR2$gene.SNP!=&quot;ENSG00000087586.rs5842156&quot;&amp;
               EUR2$gene.SNP!=&quot;ENSG00000171522.rs139036988&quot;&amp;
               EUR2$gene.SNP!=&quot;ENSG00000196284.rs112143344&quot;&amp;
               EUR2$gene.SNP!=&quot;ENSG00000135316.rs71553453&quot;&amp;
               EUR2$gene.SNP!=&quot;ENSG00000232559.rs377632592&quot;&amp;
               EUR2$gene.SNP!=&quot;ENSG00000131558.rs141161799&quot;&amp;
               EUR2$gene.SNP!=&quot;ENSG00000162441.rs36126617&quot;&amp;
               EUR2$gene.SNP!=&quot;ENSG00000162441.rs151178549&quot;&amp;
               EUR2$gene.SNP!=&quot;ENSG00000142794.rs35506192&quot;&amp;
               EUR2$gene.SNP!=&quot;ENSG00000142794.rs145038894&quot;&amp;
               EUR2$gene.SNP!=&quot;ENSG00000116128.rs10657777&quot;&amp;
               EUR2$gene.SNP!=&quot;ENSG00000117280.rs149256505&quot;&amp;
               EUR2$gene.SNP!=&quot;ENSG00000107719.rs138584752&quot;&amp;
               EUR2$gene.SNP!=&quot;ENSG00000198561.rs142865693&quot;&amp;
               EUR2$gene.SNP!=&quot;ENSG00000087365.rs5792377&quot;&amp;
               EUR2$gene.SNP!=&quot;ENSG00000162341.rs111786372&quot;&amp;
               EUR2$gene.SNP!=&quot;ENSG00000110092.rs59333593&quot;&amp;
               EUR2$gene.SNP!=&quot;ENSG00000111215.rs61604574&quot;&amp;
               EUR2$gene.SNP!=&quot;ENSG00000165502.rs141634854&quot;&amp;
               EUR2$gene.SNP!=&quot;ENSG00000104093.rs138911097&quot;)


YRI1&lt;-subset(YRI2, YRI2$gene.SNP!=&quot;ENSG00000143106.rs545041240&quot;&amp;
               YRI2$gene.SNP!=&quot;ENSG00000078403.rs528760884&quot;&amp;
               YRI2$gene.SNP!=&quot;ENSG00000167996.rs150035626&quot;&amp;
               YRI2$gene.SNP!=&quot;ENSG00000111252.rs367797687&quot;&amp;
               YRI2$gene.SNP!=&quot;ENSG00000172458.rs112879834&quot;&amp;
               YRI2$gene.SNP!=&quot;ENSG00000183044.rs367630500&quot;&amp; 
               YRI2$gene.SNP!=&quot;ENSG00000154874.rs138555657&quot;&amp;
               YRI2$gene.SNP!=&quot;ENSG00000011132.rs71166969&quot;&amp;
               YRI2$gene.SNP!=&quot;ENSG00000125991.rs142898689&quot;&amp;
               YRI2$gene.SNP!=&quot;ENSG00000198832.rs35065681&quot;&amp;
               YRI2$gene.SNP!=&quot;ENSG00000128268.rs142897975&quot;&amp;
               YRI2$gene.SNP!=&quot;ENSG00000161013.rs71591436&quot;&amp;
               YRI2$gene.SNP!=&quot;ENSG00000170727.rs60257564&quot;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.1 (2023-06-16)
Platform: x86_64-apple-darwin20 (64-bit)
Running under: macOS Sonoma 14.6.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: Europe/Oslo
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] workflowr_1.7.1

loaded via a namespace (and not attached):
 [1] vctrs_0.6.5       httr_1.4.7        cli_3.6.2         knitr_1.45       
 [5] rlang_1.1.3       xfun_0.42         stringi_1.8.3     processx_3.8.3   
 [9] promises_1.3.0    jsonlite_1.8.8    glue_1.7.0        rprojroot_2.0.4  
[13] git2r_0.33.0      htmltools_0.5.7   httpuv_1.6.15     ps_1.7.6         
[17] sass_0.4.8        fansi_1.0.6       rmarkdown_2.26    jquerylib_0.1.4  
[21] tibble_3.2.1      evaluate_0.23     fastmap_1.1.1     yaml_2.3.8       
[25] lifecycle_1.0.4   whisker_0.4.1     stringr_1.5.1     compiler_4.3.1   
[29] fs_1.6.3          pkgconfig_2.0.3   Rcpp_1.0.12       rstudioapi_0.15.0
[33] later_1.3.2       digest_0.6.34     R6_2.5.1          utf8_1.2.4       
[37] pillar_1.9.0      callr_3.7.5       magrittr_2.0.3    bslib_0.6.1      
[41] tools_4.3.1       cachem_1.0.8      getPass_0.2-4    </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
