<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="mariesaitou" />

<meta name="date" content="2020-08-13" />

<title>first-analysis</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">myproject</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/mariesaitou/myproject">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">first-analysis</h1>
<h4 class="author">mariesaitou</h4>
<h4 class="date">2020-08-13</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-08-30
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>myproject/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200813code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200813)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200813code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200813)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcommariesaitoumyprojecttreebf3a8b69ae0a47dca4969cd41d46e529099052dctargetblankbf3a8b6a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/mariesaitou/myproject/tree/bf3a8b69ae0a47dca4969cd41d46e529099052dc" target="_blank">bf3a8b6</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcommariesaitoumyprojecttreebf3a8b69ae0a47dca4969cd41d46e529099052dctargetblankbf3a8b6a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/mariesaitou/myproject/tree/bf3a8b69ae0a47dca4969cd41d46e529099052dc" target="_blank">bf3a8b6</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    analysis/.DS_Store
    Ignored:    genotype/

Untracked files:
    Untracked:  ALL.chr1.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz
    Untracked:  ALL.chr2.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz
    Untracked:  getfastqtest2.csv

Unstaged changes:
    Deleted:    analysis/first-analysis.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/prep.Rmd</code>) and HTML (<code>docs/prep.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mariesaitou/myproject/d55436d8ec7c90e6c2c3d233ee1fe3d81e22c723/docs/prep.html" target="_blank">d55436d</a>
</td>
<td>
mariesaitou
</td>
<td>
2020-08-30
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mariesaitou/myproject/d81e9a368b82ff59532766d6c1b9da41be524729/docs/prep.html" target="_blank">d81e9a3</a>
</td>
<td>
mariesaitou
</td>
<td>
2020-08-30
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mariesaitou/myproject/7c2b7b46cd403e6d48051c00e15f2349176ca720/docs/prep.html" target="_blank">7c2b7b4</a>
</td>
<td>
mariesaitou
</td>
<td>
2020-08-30
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mariesaitou/myproject/898542f5689a6975218e52512e2ed7f8145e122f/docs/prep.html" target="_blank">898542f</a>
</td>
<td>
mariesaitou
</td>
<td>
2020-08-30
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mariesaitou/myproject/1c78096f0c8dddd1809b15f78514045e227de7ad/docs/prep.html" target="_blank">1c78096</a>
</td>
<td>
mariesaitou
</td>
<td>
2020-08-30
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/mariesaitou/myproject/3a2eead3edfc3e5cc4107c525c3914b89267d7a6/docs/prep.html" target="_blank">3a2eead</a>
</td>
<td>
mariesaitou
</td>
<td>
2020-08-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/mariesaitou/myproject/blob/8a5d7d1951d4898e3afcb639e88f9b4abb117594/analysis/prep.Rmd" target="_blank">8a5d7d1</a>
</td>
<td>
mariesaitou
</td>
<td>
2020-08-16
</td>
<td>
wflow_publish(c(“analysis/index.Rmd”, “analysis/correl.Rmd”,
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="section" class="section level26">
<p></p>
<p>The master working directory on UChicago RCC</p>
<pre class="bash"><code>cd /project2/xuanyao/marie/E-GEUV-1</code></pre>
</div>
<div id="download-the-gene-expression-data-and-genotype-files" class="section level1">
<h1>Download the gene expression data and genotype files</h1>
<pre class="bash"><code>
 # Download the fastq dataset from the Geuvadis project
 # sample information https://www.ebi.ac.uk/arrayexpress/files/E-GEUV-1/
 # fastq files ftp://ftp.sra.ebi.ac.uk/vol1/fastq/
 
cat getfastq.csv | awk &#39;NR==21,NR==90&#39; &gt; getfastqtest2.csv
csvfile=getfastqtest2.csv
for line in `cat ${csvfile} | grep -v ^#`
do
  url=`echo ${line} | cut -d &#39;,&#39; -f 2`
  file=`echo ${line} | cut -d &#39;,&#39; -f 4`
  wget ${url} -O fastq2/${file}.fastq.gz
done</code></pre>
<p>Download the genotype dataset from the 1000 Genomes project phase 3</p>
<pre class="bash"><code>mkdir genotype
wget ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/release/20130502/ALL.chr{1..22}.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz</code></pre>
</div>
<div id="gene-expression-quantification-with-kallisto" class="section level1">
<h1>Gene expression quantification with Kallisto</h1>
<p>Kallisto - make index files</p>
<pre class="bash"><code>module load kallisto
wget ftp://ftp.ensembl.org/pub/grch37/current/fasta/homo_sapiens/cdna/Homo_sapiens.GRCh37.cdna.all.fa.gz
gunzip Homo_sapiens.GRCh37.cdna.all.fa.gz
kallisto index -i human.GRCh37.cdna.all.idx Homo_sapiens.GRCh37.cdna.all.fa

wget https://github.com/pachterlab/kallisto-transcriptome-indices/releases/download/ensembl-96/homo_sapiens.tar.gz
gunzip Homo_sapiens
homo_sapiens/transcriptome.idx</code></pre>
<p>Gene expression quantification for all samples with Kallisto</p>
<pre class="bash"><code>module load kallisto
csvfile=sample.filelist.csv
for line in `cat ${csvfile} | grep -v ^#`
do
  file=`echo ${line} | cut -d &#39;,&#39; -f 5`
  kallisto quant -i human.GRCh37.cdna.all.idx -o kallisto/${file}.kallistoOut -n 100 -t 32 fastq/${file}.1.fastq fastq/${file}.2.fastq
done
</code></pre>
<pre class="r"><code># summarize TPM (gene expression) data from each individual
samples &lt;-read.csv(&quot;1000g.EUR.sample.csv&quot;,header=TRUE)
files &lt;- file.path(dir, &quot;kallisto&quot;, samples$kallisto, &quot;abundance.tsv&quot;)
names(files) &lt;- paste0(samples$X) 
all(file.exists(files))
txi &lt;- tximport(files, type = &quot;kallisto&quot;, tx2gene = tx2gene, txOut = FALSE, ignoreAfterBar = TRUE)
write.table(txi$abundance,sep=&quot;\t&quot;, file = &quot;EUR.TPM.tsv&quot;)</code></pre>
<pre class="r"><code># add gene annotations to the TPM file
library(&#39;biomaRt&#39;)
mart &lt;- useDataset(&quot;hsapiens_gene_ensembl&quot;, useMart(&quot;ensembl&quot;, host=&quot;grch37.ensembl.org&quot;))
df &lt;- read.csv(file.choose(), header = T, sep = &quot;,&quot;) 
gene &lt;- df$gene
G_list &lt;- getBM(filters= &quot;ensembl_gene_id&quot;, attributes=c(&#39;ensembl_gene_id&#39;, &#39;hgnc_symbol&#39;,&#39;description&#39;,&#39;chromosome_name&#39;,&#39;start_position&#39;,&#39;end_position&#39;),mart= mart,values=gene)
G_list2&lt;-merge(df,G_list,by.x=&quot;gene&quot;,by.y=&quot;ensembl_gene_id&quot;,all.x=T)
write.csv(G_list2, file = &quot;British.gene.TPM.csv&quot;)</code></pre>
</div>
<div id="clean-the-genotype-files-with-vcftools" class="section level1">
<h1>Clean the genotype files with vcftools</h1>
<pre class="bash"><code># keep the individuals who are reported in both the 1000 Genome Project and Geuvadis
module load vcftools
for i in `seq 1 22`
do
   vcftools --gzvcf /genotype/phase3/ALL.chr$i.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz --keep 1000g.ind.sample.txt --recode --out /genotype/phase3/chr$i.1000gphase3.455 
done

# exclude rare variants 
vcftools --vcf /genotype/phase3/chr$i.1000gphase3.455.recode.vcf --maf 0.01 --max-maf 0.99 --recode --out /genotype/phase3/chr$i.1000gphase3.455.0.01

# keep only bi-allelic variants
vcftools --vcf /genotype/phase3/chr$i.1000gphase3.455.0.01.recode.vcf --min-alleles 2 --max-alleles 2 --recode --out /genotype/phase3/chr$i.1000gphase3.455.0.01.biallelic</code></pre>
</div>
<div id="identify-covariates-genetic-and-non-genetic" class="section level1">
<h1>Identify covariates (genetic and non-genetic)</h1>
<pre class="bash"><code># Convert vcf file to plink format
plink --vcf chr22.1000gphase3.455.0.01.recode.vcf --make-bed --out chr22.1000gphase3.455.0.01.recode.bed

# Run pca
plink --bfile chr22.1000gphase3.455.0.01.recode.bed --pca --out

# Plot pca
library(ggplot2)
ggplot(data=pca, aes(V3,V4)) + geom_point()
</code></pre>
<pre class="r"><code>#  surrogate variable analysis
library(sva)
mm &lt;- model.matrix(~ population, colData(ddsTxi))
mm0 &lt;- model.matrix(~ 1, colData(ddsTxi))
norm.cts &lt;- norm.cts[rowSums(norm.cts) &gt; 10,]
fit &lt;- svaseq(norm.cts, mod=mm, mod0=mm0)
# 50 SVs are found. I use 7 SVs
sva7 = sva(norm.cts , mm, mm0, n.sv=7)
write.csv(sva7$sv[,1:7], file = &quot;sva7.csv&quot;)</code></pre>
</div>
<div id="differentially-expressed-gene-analysis" class="section level1">
<h1>Differentially Expressed Gene Analysis</h1>
<pre class="r"><code>library(&quot;DESeq2&quot;)
colData &lt;-read.csv(&quot;1000g.sample.csv&quot;,header = T,row.names=1)
class(colData$date)
colData$date1&lt;-factor(colData$date)


ddsTxi &lt;- DESeqDataSetFromTximport(txi,
                                   colData = colData,
                                   design = ~ sex+population+performer)

colData &lt;-read.csv(&quot;1000g.EUR.YRI.csv&quot;,header = T,row.names=1)

ddsEurYri &lt;- DESeqDataSetFromTximport(txi,
                                   colData = colData,
                                   design = ~ sex+population+V1+V2+V3+V4+V5+V6+V7)

keep &lt;- rowsume(counds(ddsEurYri)) &gt;=455
ddsEurYri &lt;- ddsEurYri[keep,]
ddsEY&lt;-DESeq(ddsEurYri)
deg &lt;- results(ddsEY, contrast=c(&quot;population&quot;,&quot;EUR&quot;,&quot;Yoruba&quot;))
write.csv(deg, file = &quot;E-GEUV_EUR_Yoruba.csv&quot;)</code></pre>
</div>
<div id="for-fastqtl" class="section level1">
<h1>for fastQTL</h1>
<p>preparation</p>
<pre class="bash"><code># prepare input bed.gz and index files
module load bedtools
bedtools sort -i British.gene.TPM.bed -header &gt; British.gene.TPM.sorted.bed
bgzip British.gene.TPM.sorted.bed &amp;&amp; tabix -p bed British.gene.TPM.sorted.bed.gz


# prepare input vcf.gz and index files
module load htslib 
module load tabix
for i in `seq 1 22`
do
    bgzip genotype/phase3/chr$i.1000gphase3.Yoruba.0.01.biallelic.recode.vcf  &amp;&amp; tabix -p vcf genotype/phase3/chr$i.1000gphase3.Yoruba.0.01.biallelic.recode.vcf.gz
done

for i in `seq 1 22`
do
    bgzip genotype/phase3/chr$i.1000gphase3.EUR.0.01.biallelic.recode.vcf  &amp;&amp; tabix -p vcf genotype/phase3/chr$i.1000gphase3.EUR.0.01.biallelic.recode.vcf.gz
done
</code></pre>
<p>run fastQTL</p>
<pre class="bash"><code>## https://github.com/francois-a/fastqtl

for i in `seq 1 22`
do
  ./bin/fastQTL.static fastQTL --vcf /project2/xuanyao/marie/E-GEUV-1/genotype/phase3/chr$i.1000gphase3.Yoruba.0.01.biallelic.recode.vcf.gz --bed GEUV/Yoruba.gene.TPM.sorted.bed.gz --region $i:1-249250621  --threshold 0.01 --out GEUV/Yoruba.chr$i.nominals.0.01.txt --cov GEUV/Yoruba.cov.txt --normal
done
cat Yoruba.chr*.nominals.0.01.txt &gt; Yoruba.auto.nominals.0.01.txt
</code></pre>
</div>
<div id="cleaning-and-formatting-the-eqtl-result" class="section level1">
<h1>cleaning and formatting the eQTL result</h1>
<pre class="r"><code>setwd(&quot;/project2/xuanyao/marie/E-GEUV-1&quot;)
library(ggplot2)
library(&quot;ggrepel&quot;)
library(&quot;plyr&quot;)
library(gplots)
library(reshape2)


## remove multi-allelic SNPs reported as bi-allelic SNPs which confuse the result
EUR1&lt;-subset(EUR2, 
               EUR2$gene.SNP!=&quot;ENSG00000188659.rs542232278&quot;&amp;
               EUR2$gene.SNP!=&quot;ENSG00000255769.rs145926341&quot;&amp;
               EUR2$gene.SNP!=&quot;ENSG00000255769.rs371891811&quot;&amp;
               EUR2$gene.SNP!=&quot;ENSG00000259328.rs145926341&quot;&amp;
               EUR2$gene.SNP!=&quot;ENSG00000259323.rs145926341&quot;&amp;
               EUR2$gene.SNP!=&quot;ENSG00000259472.rs145926341&quot;&amp;
               EUR2$gene.SNP!=&quot;ENSG00000103942.rs1610794&quot;&amp;
               EUR2$gene.SNP!=&quot;ENSG00000103342.rs140839133&quot;&amp;
               EUR2$gene.SNP!=&quot;ENSG00000159202.rs77094622&quot;&amp;
               EUR2$gene.SNP!=&quot;ENSG00000189050.rs112549034&quot;&amp;
               EUR2$gene.SNP!=&quot;ENSG00000108592.rs138776605&quot;&amp;
               EUR2$gene.SNP!=&quot;ENSG00000256771.rs10650867&quot;&amp;
               EUR2$gene.SNP!=&quot;ENSG00000134330.rs139337028&quot;&amp;
               EUR2$gene.SNP!=&quot;ENSG00000125991.rs139252705&quot;&amp;
               EUR2$gene.SNP!=&quot;ENSG00000214078.rs142898689&quot;&amp;               
               EUR2$gene.SNP!=&quot;ENSG00000087586.rs5842156&quot;&amp;
               EUR2$gene.SNP!=&quot;ENSG00000171522.rs139036988&quot;&amp;
               EUR2$gene.SNP!=&quot;ENSG00000196284.rs112143344&quot;&amp;
               EUR2$gene.SNP!=&quot;ENSG00000135316.rs71553453&quot;&amp;
               EUR2$gene.SNP!=&quot;ENSG00000232559.rs377632592&quot;&amp;
               EUR2$gene.SNP!=&quot;ENSG00000131558.rs141161799&quot;&amp;
               EUR2$gene.SNP!=&quot;ENSG00000162441.rs36126617&quot;&amp;
               EUR2$gene.SNP!=&quot;ENSG00000162441.rs151178549&quot;&amp;
               EUR2$gene.SNP!=&quot;ENSG00000142794.rs35506192&quot;&amp;
               EUR2$gene.SNP!=&quot;ENSG00000142794.rs145038894&quot;&amp;
               EUR2$gene.SNP!=&quot;ENSG00000116128.rs10657777&quot;&amp;
               EUR2$gene.SNP!=&quot;ENSG00000117280.rs149256505&quot;&amp;
               EUR2$gene.SNP!=&quot;ENSG00000107719.rs138584752&quot;&amp;
               EUR2$gene.SNP!=&quot;ENSG00000198561.rs142865693&quot;&amp;
               EUR2$gene.SNP!=&quot;ENSG00000087365.rs5792377&quot;&amp;
               EUR2$gene.SNP!=&quot;ENSG00000162341.rs111786372&quot;&amp;
               EUR2$gene.SNP!=&quot;ENSG00000110092.rs59333593&quot;&amp;
               EUR2$gene.SNP!=&quot;ENSG00000111215.rs61604574&quot;&amp;
               EUR2$gene.SNP!=&quot;ENSG00000165502.rs141634854&quot;&amp;
               EUR2$gene.SNP!=&quot;ENSG00000104093.rs138911097&quot;)


YRI1&lt;-subset(YRI2, YRI2$gene.SNP!=&quot;ENSG00000143106.rs545041240&quot;&amp;
               YRI2$gene.SNP!=&quot;ENSG00000078403.rs528760884&quot;&amp;
               YRI2$gene.SNP!=&quot;ENSG00000167996.rs150035626&quot;&amp;
               YRI2$gene.SNP!=&quot;ENSG00000111252.rs367797687&quot;&amp;
               YRI2$gene.SNP!=&quot;ENSG00000172458.rs112879834&quot;&amp;
               YRI2$gene.SNP!=&quot;ENSG00000183044.rs367630500&quot;&amp; 
               YRI2$gene.SNP!=&quot;ENSG00000154874.rs138555657&quot;&amp;
               YRI2$gene.SNP!=&quot;ENSG00000011132.rs71166969&quot;&amp;
               YRI2$gene.SNP!=&quot;ENSG00000125991.rs142898689&quot;&amp;
               YRI2$gene.SNP!=&quot;ENSG00000198832.rs35065681&quot;&amp;
               YRI2$gene.SNP!=&quot;ENSG00000128268.rs142897975&quot;&amp;
               YRI2$gene.SNP!=&quot;ENSG00000161013.rs71591436&quot;&amp;
               YRI2$gene.SNP!=&quot;ENSG00000170727.rs60257564&quot;)</code></pre>
<p>Manhattan plot example</p>
<pre class="r"><code>x&lt;-read.csv(&quot;YRI.EUR.DEG.for.manhattan.csv&quot;, header = TRUE)
x&lt;- x[order(x$Chromosome),]
nCHR &lt;- length(unique(x$Chromosome))
x$BPcum &lt;- NA
s &lt;- 0
nbp &lt;- c()

for (i in unique(x$Chromosome)){
  nbp[i] &lt;- max(x[x$Chromosome == i,]$start)
  x[x$Chromosome == i,&quot;BPcum&quot;] &lt;- x[x$Chromosome == i,&quot;start&quot;] + s
  s &lt;- s + nbp[i]
}


axis.set &lt;- x %&gt;% 
  group_by(Chromosome) %&gt;% 
  summarize(center = (max(BPcum) + min(BPcum)) / 2)

x1 &lt;- subset (x, -log10(padj)&gt;40 )


manhplot &lt;- ggplot(x, aes(x = BPcum, y = sign(log2FoldChange)*(-log10(padj)), 
                          color = as.factor(Chromosome))) +
  geom_point(alpha = 0.85)  + 
  scale_x_continuous(label = axis.set$Chromosome, breaks = axis.set$center)  +
  scale_color_manual(values = rep(c(&quot;gray&quot;, &quot;black&quot;), nCHR)) +
  scale_size_continuous(range = c(0.5,3))  + 
  theme_minimal() +theme_bw(base_size = 12)+
  theme( 
    legend.position = &quot;none&quot;,
    panel.border = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank(),
    axis.text.x = element_text(angle = 90, size = 8, vjust = 0.5)
  )+geom_label_repel(data=x1,aes(label = gene.name), color = &#39;blue&#39;,
                     size = 3) </code></pre>
<pre class="r"><code># GTEx comparison
# GTEx.Cells_EBV-transformed_lymphocytes.v8.egenes.txt
# GTEx_Analysis_2017-06-05_v8_RNASeQCv1.1.9_gene_median_tpm.gct</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.0 (2020-04-24)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Catalina 10.15.6

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] workflowr_1.6.2

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.4.6    rprojroot_1.3-2 digest_0.6.25   later_1.1.0.1  
 [5] R6_2.4.1        backports_1.1.8 git2r_0.27.1    magrittr_1.5   
 [9] evaluate_0.14   stringi_1.4.6   rlang_0.4.7     fs_1.4.1       
[13] promises_1.1.1  whisker_0.4     rmarkdown_2.3   tools_4.0.0    
[17] stringr_1.4.0   glue_1.4.1      httpuv_1.5.4    xfun_0.14      
[21] yaml_2.2.1      compiler_4.0.0  htmltools_0.4.0 knitr_1.28     </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
